### PR TITLE
feat: enable SecurityContext option for eventbus pods. Closes #903

### LINKED ISSUE #903: Argo EventBus runAsUser configuration
**Describe the bug**
The event bus controller runs as user configured by me (999) . But the eventbus resource generated fails for below. Is there a way to configure the runasuser (pod securitycontext)  ?

apiVersion: argoproj.io/v1alpha1
kind: EventBus
metadata:
  name: default
spec:
  nats:
    native:
      # Optional, defaults to 3. If it is < 3, set it to 3, that is the minimal requirement.
      replicas: 3
      # Optional, authen strategy, "none" or "token", defaults to "none"
      auth: token

Events:
  Type     Reason     Age              From                                    Message
  ----     ------     ----             ----                                    -------
  Normal   Scheduled  5s               default-scheduler                       Successfully assigned argo-events/eventbus-default-stan-0 to ip-100-72-10-188.ec2.internal
  Normal   Pulled     4s (x2 over 4s)  kubelet, ip-100-72-10-188.ec2.internal  Container image "nats-streaming:0.17.0@sha256:be42a6dcfd820c1605b5f0ad5f99997eb1df73fca192fbde8a163a0bc5c62131" already present on machine
  Warning  Failed     4s (x2 over 4s)  kubelet, ip-100-72-10-188.ec2.internal  **Error: container has runAsNonRoot and image will run as root**
  Normal   Pulled     4s (x2 over 4s)  kubelet, ip-100-72-10-188.ec2.internal  Container image "synadia/prometheus-nats-exporter:0.6.2@sha256:e6a6cfe30f3cbae8bad1e8dd6f6c68f4401e9b563aeb2df73769f1b4dad39a0f" already present on machine
  Warning  Failed     4s (x2 over 4s)  kubelet, ip-100-72-10-188.ec2.internal  **Error: container has runAsNonRoot and image will run as root**

### PR DESCRIPTION
**Pull request by whynowy**
Checklist:

* [x] Either (a) I've created an [feature request](https://github.com/argoproj/argo-events/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

Closes #903

### CODE REVIEW

File: controllers/eventbus/installer/nats.go [hunk 1]
```diff
@@ -411,39 +411,35 @@ func (i *natsInstaller) createStatefulSet(ctx context.Context, serviceName, conf
...
-				log.Desugar().Error("error updating statefulset", zap.Error(err))
-				return err
-			}
-			log.Infow("statefulset is updated", "statefulsetName", ss.Name)
+		if ss.Annotations != nil && ss.Annotations[common.AnnotationResourceSpecHash] == expectedSs.Annotations[common.AnnotationResourceSpecHash] {
+			return nil
 		}
-	} else {
-		err := i.client.Create(ctx, expectedSs)
+		// Delete the existing one to recreate it
```
Comment (whynowy) [inline controllers/eventbus/installer/nats.go:427]: Change update logic to delete and create, this is better for StatefulSet.

**Comment by mailtoraja18** [comment]
Do we need to fix EventSource and Sensor pods as well to pass in SecurityContext. Thanks

**Comment by whynowy** [comment]
> Do we need to fix EventSource and Sensor pods as well to pass in SecurityContext. Thanks

EventSource and Sensor already have that.

**Review by VaibhavPage — APPROVED** [review]
lgtm

**Pull request merged**